components/layout/global-header.tsx

```diff
@@ -1,16 +1,42 @@
-import { createElement, useRef } from "react";
+import { createElement, useRef, useState } from "react";
 
 import { Popover } from "@headlessui/react";
 import Link from "next/link";
 import { Container } from "../ui/container";
+import useDocumentScrollThrottled from "../ui/scroll-throttled";
 import { NavigationMenuPopover, navMenuButtonIcon } from "./navigation-menu";
 import { WBWLogoWhite } from "~/components/ui/wbw-logo";
 
 export function GlobalHeader() {
   const popoverButtonRef = useRef<HTMLButtonElement>(null);
 
+  const [shouldHideHeader, setShouldHideHeader] = useState(false);
+  const [shouldShowShadow, setShouldShowShadow] = useState(false);
+
+  const MINIMUM_SCROLL = 80;
+  const TIMEOUT_DELAY = 400;
+
+  useDocumentScrollThrottled(
+    (callbackData: { previousScrollTop: Number; currentScrollTop: Number }) => {
+      const { previousScrollTop, currentScrollTop } = callbackData;
+      const isScrolledDown = previousScrollTop < currentScrollTop;
+      const isMinimumScrolled = currentScrollTop > MINIMUM_SCROLL;
+
+      setShouldShowShadow(currentScrollTop > 2);
+
+      setTimeout(() => {
+        setShouldHideHeader(isScrolledDown && isMinimumScrolled);
+      }, TIMEOUT_DELAY);
+    },
+  );
```

Comment:
Please add more tests to cover these new cases that you're introducing, thanks!

Reply:
@zainfathoni do you have any references where I can start making a test for this? thank you